src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -317,11 +317,10 @@ private static void DeserializeIntoObjectProperties(this CodeWriter writer, IEnu
                              property.ValueType?.Equals(typeof(string)) !=
                              true) //https://github.com/Azure/autorest.csharp/issues/922
                     {
-                        if (Configuration.AzureArm && property.ValueType?.Equals(typeof(Uri)) == true)
+                        if (!property.IsClientFlattenedProperty) // if flatten property, then we should not try to assign null/default to parent property
```

Comment:
It is better to use `property.ValueSerialization != null` here. That way it has meaning - value of the property can't be deserialized directly (reason is not important).

Reply:
expanded the method